compiler/src/dotty/tools/dotc/transform/init/Objects.scala

```diff
@@ -77,7 +112,7 @@ object Objects:
     valsMap: mutable.Map[Symbol, Value],
     varsMap: mutable.Map[Symbol, Heap.Addr],
     outersMap: mutable.Map[ClassSymbol, Value])
-  extends Value:
+  extends ValueElement with ThisValue:
     protected val vals: mutable.Map[Symbol, Value] = valsMap
     protected val vars: mutable.Map[Symbol, Heap.Addr] = varsMap
     protected val outers: mutable.Map[ClassSymbol, Value] = outersMap
```

Comment:
Is it possible to make the value for outers more precise, e.g., `ThisValue`?

Aside: It now seems to me that `ThisValue` is not a great name. /cc: @olhotak @q-ata

Reply:
I think `outers` can be `Bottom` for top-level classes, so we cannot type it as `ThisValue` under current design